clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -2203,13 +2207,6 @@ mlir::Value ScalarExprEmitter::VisitAbstractConditionalOperator(
     }
   }
 
-  // OpenCL: If the condition is a vector, we can treat this condition like
-  // the select function.
-  if ((CGF.getLangOpts().OpenCL && condExpr->getType()->isVectorType()) ||
-      condExpr->getType()->isExtVectorType()) {
```

Comment:
Why are you removing this check? I don't think you have testcases to back this up and doing this will probably lead to some mis-compilation that is going hard to track later. There's specific handling that needs to be done for `VisitAbstractConditionalOperator` in face of these conditions.

Reply:
`vector_ternary_lowering` is on this path. This was mentioned in the PR description (I'll update it to a brief version though):

> [A special method](https://github.com/llvm/clangir/blob/3da10fafac66ff125fb59c602e41ad4b4f5cb382/clang/lib/CodeGen/CGExprScalar.cpp#L5165) is used for the original CodeGen for ternary operator of OpenCL vectors. I don't know why. Since we already have high-level cir.vec.ternary, I skipped this part. And the lowering still works well. Maybe it's about the SPIR-V target.

Probably I'll also comment the corresponding part in the PR review ahead of time : )

Reply:
> Since we already have high-level cir.vec.ternary, I skipped this part

Unfortunately that's not a good rationale and not how we usually go about these things in CIR - if this leads to a miscompile, it's going to be really hard to pinpoint the issue, and might take a lot of investigation time. If something was implemented that is shortcuting here, it should also be fixed to maintain the skeleton the much pristine as possible. It's possible you might find out that you need to do another PR to fix that problem before you can land this one.

Alternatively, if you are deleting this code so one of your testcases may pass, you can then choose to not add that testcase and address the whole issue together in another PR.